Add Azure MySQL connector with test cases (#1)
ballerina-platform/module-ballerinax-azure.mysql: Package.md

## Package overview

This Package provides the functionality required to access and manipulate data stored in a Azure MySQL database.

> **Note:** The MySQL driver JAR is included in this connector. So you don't need to add MySQL driver JAR dependency to

Reviewer: ```suggestion
**Note:** The MySQL driver JAR is included in this connector. Therefore, you don't need to add the MySQL driver JAR dependency to
```

>  the project `Ballerina.toml` file.

Reviewer: ```suggestion
 the `Ballerina.toml` file of your project.
```

## Compatibility

Ballerina Language Version   **Swan Lake Alpha 5**  
MySQL Driver Version         **8.0.20**

### Client

> To access a database, you must first create a mysql:Client object. 

Reviewer: ```suggestion
To access a database, you must first create a `mysql:Client` object. 
```

The examples for creating a MySQL client can be found below.

#### Creating a client
This example shows different ways of creating the `mysql:Client`. 

The client can be created with an empty constructor and hence, the client will be initialized with the default properties. 
The first example with the `dbClient1` demonstrates this.

> The `dbClient2` receives the host, username, and password. Since the properties are passed in the same order as it is defined 

Reviewer: ```suggestion
The `dbClient2` receives the host, username, and password. Since the properties are passed in the same order as defined 
```

> in the `jdbc:Client`, you can pass it without named params.

Reviewer: ```suggestion
in the `jdbc:Client`, you can pass them without named params.
```

The `dbClient3` uses the named params to pass the attributes since it is skipping some params in the constructor. 

> Further [mysql:Options](https://ballerina.io/learn/api-docs/ballerina/#/mysql/records/Options) 

Reviewer 2: AFAIK, these url links are not working. Should we change those links here?

Author: I will check and remove

> Further [mysql:Options](https://ballerina.io/learn/api-docs/ballerina/#/mysql/records/Options) 

Reviewer: ```suggestion
Further, the [`mysql:Options`](https://ballerina.io/learn/api-docs/ballerina/#/mysql/records/Options) 
```

> is passed to configure the SSL and connection timeout in the MySQL client. 

Reviewer: ```suggestion
property is passed to configure the SSL and connection timeouts in the MySQL client. 
```

Similarly, the `dbClient4` uses the named params and it provides an unshared connection pool in the type of 

> [sql:ConnectionPool](https://ballerina.io/learn/api-docs/ballerina/#/sql/records/ConnectionPool) 

Reviewer 2: These links are not working here too.

> [sql:ConnectionPool](https://ballerina.io/learn/api-docs/ballerina/#/sql/records/ConnectionPool) 

Reviewer: ```suggestion
[`sql:ConnectionPool`](https://ballerina.io/learn/api-docs/ballerina/#/sql/records/ConnectionPool) 
```

to be used within the client. 

> For more details about connection pooling, see the [SQL Package](https://ballerina.io/learn/api-docs/ballerina/#/sql).

Reviewer: ```suggestion
For more details about connection pooling, see the [`sql` package](https://ballerina.io/learn/api-docs/ballerina/#/sql).
```

```ballerina
mysql:Client|sql:Error dbClient1 = new ();
mysql:Client|sql:Error dbClient2 = new ("localhost", "rootUser", "rooPass", 
                              "information_schema", 3306);
                              
mysql:Options mysqlOptions = {
  ssl: {
    mode: mysql:SSL_PREFERRED
  },
  connectTimeout: 10
};
mysql:Client|sql:Error dbClient3 = new (user = "rootUser", password = "rootPass",
                              options = mysqlOptions);
mysql:Client|sql:Error dbClient4 = new (user = "rootUser", password = "rootPass",
                              connectionPool = {maxOpenConnections: 5});
```
You can find more details about each property in the

> [mysql:Client](https://ballerina.io/learn/api-docs/ballerina/#/mysql/clients/Client) constructor. 

Reviewer: ```suggestion
[`mysql:Client`](https://ballerina.io/learn/api-docs/ballerina/#/mysql/clients/Client) constructor. 
```

> The [mysql:Client](https://ballerina.io/learn/api-docs/ballerina/#/mysql/clients/Client) references 

Reviewer: ```suggestion
The [`mysql:Client`](https://ballerina.io/learn/api-docs/ballerina/#/mysql/clients/Client) references 
```

[sql:Client](https://ballerina.io/learn/api-docs/ballerina/#/sql/abstractObjects/Client) and all the operations 
defined by the `sql:Client` will be supported by the `mysql:Client` as well. 

> For more information on all the operations supported by the `mysql:Client`, which include the below,

Reviewer: ```suggestion
For more information on all the operations supported by the `mysql:Client`, which include the below, see[]().
```
The link is missing here.
